webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -82,6 +88,10 @@ export class ContactsEffects {
             console.error('Error selecting contact', err);
             this.globalActions.unsetSelected();
             return of(this.contactsActions.clearSelection());
+          })
+          .finally(() => {
+            trackPerformance?.setName(this.selectContactTrack);
+            trackPerformance?.stop(true);
```

Comment:
It's really weird to set the name on creation and on stop. Let's change the API so we only do it on stop? eg:

```suggestion
            trackPerformance?.stop(this.selectContactTrack, true);
```

Then I think you can clean up the code even more if instead of storing the name in `this.selectContactTrack` you calculate it only when you need it in the `finally` block.